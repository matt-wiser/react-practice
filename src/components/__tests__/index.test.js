import React from "react";
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import About from "../About";
import Nav from "../Nav";

afterEach(cleanup);

describe("About component", () => {
    // First test
    it("renders", () => {
        render(<About/>);
        
    });

    // Second test
    it("Matches snapshot DOM node stucture", () => {
        const {asFragment} = render(<About/>);

        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Nav component", () => {
    // First test
    it("renders", () => {
        render(<Nav/>);
        
    });

    // Second test
    it("Matches snapshot DOM node stucture", () => {
        const {asFragment} = render(<Nav/>);

        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Emoji is visible", () => {
    it("Inserts emoji into the h2 tag", () => {
        const { getByLabelText } = render(<Nav/>);
    
        expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
    });
});

describe("Links are visible", () => {
    it("Inserts text into the link tags", () => {
        const { getByTestId } = render(<Nav/>);
    
        expect(getByTestId('link')).toHaveTextContent('Oh Snap!');
        expect(getByTestId('about')).toHaveTextContent('About Me');
    });
});