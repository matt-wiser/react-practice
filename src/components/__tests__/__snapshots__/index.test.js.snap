// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Matches snapshot DOM node stucture 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="about"
    >
      Who Am I?
    </h1>
    <img
      alt="cover"
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;

exports[`Nav component Matches snapshot DOM node stucture 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
           ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
